R/ranova.b.R

```diff
@@ -1,393 +1,1944 @@
+
+                for(i in 1:7)
+                    rl <- append(rl, private$.medpb(x,
+                                                    con=con[[i]],
+                                                    alpha=alpha,
+                                                    est=median))
+            }
+            else { # trim
+
+                for(i in 1:7)
+                    rl <- append(rl, private$.lincon(x,
+                                                     con=con[[i]],
+                                                     alpha=alpha,
+                                                     tr=self$options$tr))
+            }
+
+            return(rl)
+        },
+        .cleanData=function() {
+
+            dep<-self$options$dep
+            factors<-self$options$factors
+
+            data<-self$data
+
+            if ( ! is.null(dep))
+                data[[dep]]<-jmvcore::toNumeric(data[[dep]])
+
+            for (factor in factors)
+                data[[factor]]<-as.factor(data[[factor]])
+
+            data
+        },
+        .ff=function() {
+
+            factors<-self$options$factors
+
+            if (length(factors) > 1) {
+                formula<-as.formula(paste('~', paste(paste0('`', factors, '`'), collapse='*')))
+                terms  <-attr(stats::terms(formula), 'term.labels')
+                modelTerms<-sapply(terms, function(x) as.list(strsplit(x, ':')), USE.NAMES=FALSE)
+            } else {
+                modelTerms<-as.list(factors)
+            }
+
+            for (i in seq_along(modelTerms)) {
+                term<-modelTerms[[i]]
+                quoted<-grepl('^`.*`$', term)
+                term[quoted]<-substring(term[quoted], 2, nchar(term[quoted])-1)
+                modelTerms[[i]]<-term
+            }
+
+            modelTerms
+        },
+
+        ### --------------------------------------------------------------------
+        # Reference:
+        # A package of R. R. Wilcox' robust statistics functions.
+        # For more information, see https://dornsife.usc.edu/labs/rwilcox/software/
+        # Here the latest update of the Rallfun-v40.txt file (lic. and All func.)
+        # https://dornsife.usc.edu/assets/sites/239/docs/Rallfun-v40.txt
+        #
+        # Some of these functions are used here, with some minor changes.
+        ### --------------------------------------------------------------------
+        .fac2list=function(x, g) {
+            # data are stored in x
+            # information about the level of the value in x is stored in g,
+            # which can be a matrix with up to 4 columns sort the data in x
+            # into groups based on values in g, store results in list mode.
+
+            g=as.data.frame(g)
+            ncg=ncol(g)
+
+            y<-list()
+            for(nc in 1:ncg)
+                y[[nc]]<-as.factor(g[,nc])
+
+            # take data in list mode and store it in a matrix
+            lgy=length(y)
+            nval=NA
```

Comment:
good to be consistent ... either use `=` or `<-` ...

ideally, you'd be consistent with the rest of the module too, (i.e. with spaces around ` <- `) but you're contributing so much to this module, i don't mind you adopting your own style ... although that does mean this will be a different style to the other analyses in this module.

for future, it is good to develop the habit of maintaining the style of the codebase you're contributing to (i'm flexible here, but ravi wouldn't be :P)